src/server.rs

```diff
@@ -20,6 +20,8 @@ use crate::ClientServerMap;
 
 /// Server state.
 pub struct Server {
+    server_id: i32,
```

Comment:
This is a big change. So I am using a randomly generated number to refer to individual server connections and using that to store server stats. 

The server_stats is a global object (spans across all pools in the process) and I figured it is not impossible that two PG backend PIDs my collide, so with in this PR, `server.process_id` is used only for query cancellation. Stats will track a different number that is generated randomly like the one for the client